Source/Core/Core/AchievementManager.cpp

```diff
@@ -34,18 +34,23 @@ AchievementManager& AchievementManager::GetInstance()
 
 void AchievementManager::Init()
 {
-  if (!m_is_runtime_initialized && Config::Get(Config::RA_ENABLED))
+  if (!m_client && Config::Get(Config::RA_ENABLED))
   {
+    m_client = rc_client_create(MemoryPeeker, Request);
     std::string host_url = Config::Get(Config::RA_HOST_URL);
     if (!host_url.empty())
-      rc_api_set_host(host_url.c_str());
-    rc_runtime_init(&m_runtime);
-    m_is_runtime_initialized = true;
+      rc_client_set_host(m_client, host_url.c_str());
+    rc_client_set_event_handler(m_client, EventHandler);
+    rc_client_enable_logging(m_client, RC_CLIENT_LOG_LEVEL_VERBOSE,
+                             [](const char* message, const rc_client_t* client) {
+                               INFO_LOG_FMT(ACHIEVEMENTS, "{}", message);
+                             });
+    rc_client_set_hardcore_enabled(m_client, Config::Get(Config::RA_HARDCORE_ENABLED));
     m_queue.Reset("AchievementManagerQueue", [](const std::function<void()>& func) { func(); });
     m_image_queue.Reset("AchievementManagerImageQueue",
                         [](const std::function<void()>& func) { func(); });
     if (IsLoggedIn())
```

Comment:
This isn't a part of the this PR but `IsLoggedIn()` makes it seem like this is checking if we're already logged in, so it's then surprising to see `Login()` being called after.  This should probably be renamed `IsLoginRequired()` or something similar in the future.

Reply:
What this is actually doing is `if (HasStoredSessionToken()) { SendLoginRequestToServer(); }`, fwiw. I agree that the naming here is bad, I'm pretty sure I pointed this out in some other PR as well...